lib/config-base.d.ts

```diff
@@ -61,6 +61,11 @@ export interface HTTPOptions {
      * Used in node.js environments only.
      */
     connectTimeout?: number;
+    /**
+     * Custom DNS lookup function.
+     * Defaults to dns.lookup.
```

Comment:
Nit: can you specify that they are node.js environments only? And looks like it's only available in newer version: https://github.com/nodejs/node/commit/5bd6f516d82d069ff8710b86108dedc333b2b580

```suggestion
     * Defaults to dns.lookup.
     * Used in Node.js (>= v12.x) environment only.
```